### PR TITLE
1337 refactor about you component - About you section [WIP]

### PR DESCRIPTION
**Pull request by tavy88**
### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: app/components/about_you_component.rb [hunk 1]
```diff
@@ -8,69 +8,57 @@ class AboutYouComponent < ViewComponent::Base
...
-          text: "Your name"
-        },
-        value: {
-          text: nullable_value_to_s("#{referrer&.first_name} #{referrer&.last_name}".presence)
-        }
-      },
-      { actions: [], key: { text: "Your email address" }, value: { text: user.email } }
-    ]
+    items = [your_name, your_email_address, your_job_title, your_phone_number].compact
+
```
Comment (tavy88) [inline app/components/about_you_component.rb:12]: Or would you rather see something like:

```ruby
items.push(your_name)
items.push(your_email_address)
items.push(your_job_title) if some condition....
....
```

here?

File: app/helpers/component_helper.rb [hunk 2]
```diff
@@ -2,4 +2,20 @@ module ComponentHelper
...
+
+  def new_item(label:, value:, actions: [])
+    item_line = { key: { text: label }, value: { text: value } }
+
+    return item_line.merge(actions:) if actions.any?
+
+    item_line
+  end
+
+  def action(path:, label: "Change", options: {})
```
Comment (gpeng) [inline app/helpers/component_helper.rb:14]: I'm not sure of the value of this as it is at the moment. All it is really doing is changing the name of label to text and path to href.

File: app/components/about_you_component.rb [hunk 3]
```diff
@@ -8,69 +8,60 @@ class AboutYouComponent < ViewComponent::Base
...
+  end
+
+  def your_job_title
+    return unless referral.from_employer?
+
+    new_item(
+      actions: [
+        action(
+          path:
+            polymorphic_path([:edit, referral.routing_scope, referral, :referrer_job_title], return_to: return_to_path),
```
[the remark this answers is not in the document]
Comment (tavy88) [reply app/components/about_you_component.rb:42]: I thought I'd make it flexible enough to allow other attributes than `visually_hidden_text`.

File: app/helpers/component_helper.rb (hunk 2 above)
Comment (tavy88) [reply app/helpers/component_helper.rb:14]: Yeah thought I'd reduce the repetition of adding an action on the component level. Like stop repeating the label: Change for example.

**Comment by tavy88** [comment]
A very similar PR was created: https://github.com/DFE-Digital/refer-serious-misconduct/pull/623

**Pull request closed without merging**